core/src/org/javarosa/core/model/instance/TreeReference.java

```diff
@@ -460,7 +463,14 @@ public TreeReference contextualize(TreeReference contextRef) {
                 newRef.data.setElementAt(newRef.data.elementAt(i).setName(contextRef.getName(i)), i);
             }
 
-            if (!contextRef.getName(i).equals(newRef.getName(i))) {
+            if (contextRef.getName(i).equals(newRef.getName(i))) {
+                // Only copy over multiplicity info if it won't overwrite any
+                // existing preds or filters
+                if (newRef.getPredicate(i) == null &&
```

Comment:
Not sure if the 'newRef.getPredicate(i) == null' check is needed. Just leaving it in from the existing code.